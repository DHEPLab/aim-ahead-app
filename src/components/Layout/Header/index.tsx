import React from "react";
import { Avatar } from "@mui/material";
import DensityMediumIcon from "@mui/icons-material/DensityMedium";

import avatar from "../../../assets/images/defalt_avatar.jpg";
import styles from "./index.module.scss";

export type HeaderProps = {};

const Header: React.FC = (props: HeaderProps) => {
  return (
    <div className={styles.header}>
      <div>Collective Goods</div>
      <div className={styles.right}>
        <Avatar className={styles.avatar} src={avatar} />
        <DensityMediumIcon />
      </div>
    </div>
  );
};

export default Header;
