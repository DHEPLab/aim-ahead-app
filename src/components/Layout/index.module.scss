$mobile-point: 576px;

.layout {
  height: 100%;
  background-color: #fff;
  overflow: hidden auto;
  min-width: 320px;

  &.m{
    @media screen and (min-width: $mobile-point) {
      max-width: $mobile-point;
      margin: 0 auto;
    }
  }
  
}
