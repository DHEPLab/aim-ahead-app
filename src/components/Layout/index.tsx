import React, { useEffect, useRef } from "react";
import { useLocation } from "react-router-dom";
import classnames from "classnames";

import testId from "../../utils/testId";

import styles from "./index.module.scss";

const Layout = (props: React.PropsWithChildren<{ className?: string; style?: React.CSSProperties }>) => {
  const { pathname } = useLocation();
  const layoutRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    layoutRef.current?.scrollTo({
      top: 0,
      behavior: "smooth",
    });
  }, [layoutRef, pathname]);

  return (
    <div
      {...testId("aim-ahead-layout")}
      ref={layoutRef}
      className={classnames(
        {
          [styles.layout]: true,
          [styles.m]: window.innerWidth > 576,
        },
        props.className,
      )}
      style={props.style}
    >
      {props.children}
    </div>
  );
};

export default Layout;
