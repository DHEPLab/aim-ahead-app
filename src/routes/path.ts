export const path = {
  root: "/",
  signup: "/signup",
  login: "/login",
  resetPassword: "/reset-password/:token",
  case: "/case-review/:caseConfigId",
  answer: "/answer/:caseConfigId",
  forgotPassword: "/forgot-password",
};

export default path;
