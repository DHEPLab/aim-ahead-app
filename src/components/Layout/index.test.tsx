// eslint-disable-file testing-library/await-async-query
import React, { ReactElement } from "react";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { create } from "react-test-renderer";

import Layout from ".";
import path from "../../routes/path";

const render = (element: ReactElement) => (
  <MemoryRouter initialEntries={[path.root]}>
    <Routes>
      <Route path={path.root} element={element} />
    </Routes>
  </MemoryRouter>
);

describe("Test Layout on variable media point view port:", () => {
  const mockWindowInnerWidth = (width: number) => {
    Object.defineProperty(window, "innerWidth", {
      writable: true,
      configurable: true,
      value: width,
    });
  };

  it("should on mobile view port", async () => {
    mockWindowInnerWidth(420);

    const renderer = create(render(<Layout />));

    await renderer.root.findByProps({ className: "layout" });
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it("should on PC view port", async () => {
    mockWindowInnerWidth(1680);

    const renderer = create(render(<Layout />));

    await renderer.root.findByProps({ className: "layout m" });
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});

describe("Test Layout pass in classname", () => {
  it("Should have the default classname when no variables are passed in", async () => {
    const renderer = create(render(<Layout />));

    await renderer.root.findByProps({ className: "layout m" });
  });

  it("Should receive classname when the variables are passed in", async () => {
    const classes = "appLayout";

    const renderer = create(render(<Layout className={classes} />));

    await renderer.root.findByProps({ className: `layout m ${classes}` });
  });
});
