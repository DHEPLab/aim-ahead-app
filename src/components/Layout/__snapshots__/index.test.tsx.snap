// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Layout on variable media point view port: should on PC view port 1`] = `
<div
  className="layout m"
  data-testid="aim-ahead-layout"
  id="aim-ahead-layout"
/>
`;

exports[`Test Layout on variable media point view port: should on mobile view port 1`] = `
<div
  className="layout"
  data-testid="aim-ahead-layout"
  id="aim-ahead-layout"
/>
`;
